app/models/location.rb

```diff
@@ -168,35 +168,35 @@ def populate_detailed_hours_json_before_save
     self.detailed_hours_json = hours_info.to_json
   end
 
-  def init_detailed_hours
-    begin
-      return if self.detailed_hours_json.nil?
-    rescue ActiveModel::MissingAttributeError => e
-      return
-    end
-    detailed_hours = JSON.parse(self.detailed_hours_json)
-    return if detailed_hours.empty?
-    now = Time.new
-    @day_info = {}
-    (0..6).each do |index|
-      prefix = 'day'+index.to_s+'_'
-      next if detailed_hours[index.to_s].nil?
-      start = detailed_hours[index.to_s]['start']
-      stop = detailed_hours[index.to_s]['end']
-      next if start.nil? or stop.nil?
-      write_day_info( prefix+'status', detailed_hours[index.to_s]['status'].to_i )
-      # carefully set start time
-      t = Time.find_zone(self.time_zone).parse( start )
-      t = t.change(:year=>now.year, :month=>now.month, :day=>now.day)
-      write_day_info( prefix+'start', t )
-      # carefully set end time
-      t = Time.find_zone(self.time_zone).parse( stop )
-      t = t.change(:year=>now.year, :month=>now.month, :day=>now.day)
-      write_day_info( prefix+'end', t )
+    def init_detailed_hours
+      begin
+        return if self.detailed_hours_json.nil?
+      rescue ActiveModel::MissingAttributeError
+        return
+      end
+      detailed_hours = JSON.parse(self.detailed_hours_json)
+      return if detailed_hours.empty?
+      now = Time.new
+      @day_info = {}
+      (0..6).each do |index|
+        prefix = 'day'+index.to_s+'_'
+        next if detailed_hours[index.to_s].nil?
+        start = detailed_hours[index.to_s]['start']
+        stop = detailed_hours[index.to_s]['end']
+        next if start.nil? or stop.nil?
+        write_day_info( prefix+'status', detailed_hours[index.to_s]['status'].to_i )
+        # carefully set start time
+        t = Time.find_zone(self.time_zone).parse( start )
```

Comment:
@rylanb Nice catch! I'll pay closer attention to low hanging fruit like this next time.